src/app/campaign/store/actions/campaign-action.service.ts

```diff
@@ -77,6 +69,22 @@ export class CampaignActionService implements OnDestroy {
     return startAt && endAt && startAt.valueOf() < endAt.valueOf();
   }
 
+  hasChanged(a: any, b: any) {
```

Comment:
I was looking for a general solution to "check for form changes", and this is what I came up with.  Key diffs from a normal JSON.stringify comparison:

- "A" (the flight form) objects can be missing keys present in "B".  This is helpful for things like flight zones, where the form only has `[{id: "pre_1"}]` but the remote JSON might be `[{id: "pre_1", label: "Preroll 1", pingbacks: []}]`.
- Undefined and null are equal.  Since our HAL API tends to omit any null fields, but we end up PUT-ing back nulls from the form.

Reply:
This is an excellent improvement. Thanks for adding this.